### PR TITLE
Design create space

### PR DESCRIPTION
**Pull request by cp-ishita-g**
- [x] redesign top bar of home page
- [x] design create space screen
- [x] add api file for - space, space invitation, user
- [x] add model class for - space, invitation, location
- [x] add service file for space

https://github.com/canopas/yourspace-flutter/assets/154948719/a260c3b5-d44e-4d12-a00f-3d221d5eb9b2

### CODE REVIEW

File: app/lib/ui/flow/space/create/create_space_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    } catch (error, stack) {
+      logger.e(
+        'CreateSpaceViewNotifier: $error - error while creating new space',
+        error: error,
+        stackTrace: stack,
+      );
+    }
+  }
+
+  void updateSelectedNudgeMessage(String message) {
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/space/create/create_space_view_model.dart:38]: Give appropriate name

File: app/lib/ui/flow/space/invite/invite_code_screen.dart [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  @override
+  State<InviteCode> createState() => _InviteCodeState();
+}
+
+class _InviteCodeState extends State<InviteCode> {
+  @override
+  Widget build(BuildContext context) {
+    return PopScope(
+      onPopInvoked: (_) {
+        AppRoute.popTo(context, AppRoute.home.path);
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/space/invite/invite_code_screen.dart:32]: Is it working fine now?

File: data/lib/api/location/location.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      DocumentSnapshot<Map<String, dynamic>> snapshot,
+      SnapshotOptions? options) {
+    Map<String, dynamic>? data = snapshot.data();
+    return ApiLocation.fromJson(data!);
+  }
+
+  Map<String, dynamic> toFireStore(ApiLocation space) => space.toJson();
+}
+
+// @JsonEnum(valueField: "value")
```
Comment (cp-radhika-s) [inline data/lib/api/location/location.dart:34]: Remove this code block

File: data/lib/api/space/api_space_service.dart [hunk 4]
```diff
@@ -33,5 +54,84 @@ class ApiSpaceService {
...
+  Future<void> joinSpace(String spaceId,
+      {int role = SPACE_MEMBER_ROLE_MEMBER}) async {
+    final userId = _currentUser?.id ?? '';
+
+    final member = ApiSpaceMember(
+      space_id: spaceId,
+      user_id: userId,
+      role: role,
+      location_enabled: true,
+      id: '',
```
Comment (cp-radhika-s) [inline data/lib/api/space/api_space_service.dart:66]: The id should be the doc id

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/lib/ui/flow/space/invite/invite_code_screen.dart (hunk 2 above)
Comment (cp-ishita-g) [reply app/lib/ui/flow/space/invite/invite_code_screen.dart:32]: > Is it working fine now?

no

File: data/lib/api/space/api_space_service.dart [hunk 5]
```diff
@@ -63,7 +63,7 @@ class ApiSpaceService {
       user_id: userId,
       role: role,
       location_enabled: true,
-      id: '',
+      id: _spaceRef.doc().id,
```
Comment (cp-radhika-s) [inline data/lib/api/space/api_space_service.dart:66]: it should be space member  doc id, not space doc id,
You can set user-id as we're saving doc with userId only

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**